src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -148,6 +161,22 @@ public static bool IsWindowsDesktop
         // Gets the location for cache and config folders.
         internal static readonly string CacheDirectory = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData) + @"\Microsoft\PowerShell";
         internal static readonly string ConfigDirectory = Environment.GetFolderPath(Environment.SpecialFolder.Personal) + @"\PowerShell";
+        private static readonly Lazy<bool> _isStaSupported = new Lazy<bool>(() =>
+        {
+            // See objbase.h
+            const int COINIT_APARTMENTTHREADED = 0x2;
+            const int E_NOTIMPL = unchecked((int)0X80004001);
+            int result = Windows.NativeMethods.CoInitializeEx(IntPtr.Zero, COINIT_APARTMENTTHREADED);
+
+            // If 0 is returned the thread has been initialized for the first time
+            // as an STA and thus supported and needs to be uninitialized.
+            if (result > 0)
+            {
+                Windows.NativeMethods.CoUninitialize();
+            }
```

Comment:
I wonder why we need the check if it is used in startup scenario.
If we want to have universal code the condition looks wrong:
S_Ok = 0
S_False = 1
RPC_E_CHANGED_MODE = -2147417850

Reply:
It's needed in the startup scenario as the main console runspace needs to start as either STA or MTA.  So we need to know if STA is supported before trying STA otherwise pwsh ends up in an unusable state.  S_False will never apply here as CoInitialize() won't be called before this.